docs/sql-ref-syntax-qry-select-transform.md

```diff
@@ -57,19 +66,85 @@ SELECT TRANSFORM ( expression [ , ... ] )
+ from empty strings. This delimiter can be overridden by `NULL DEFINED AS`.
+ - Spark casts all columns to `STRING` and combine columns by tabs before feeding to the user script.
+ For complex type such as `ARRAY`/`MAP`/`STRUCT`, Spark use `to_json` cast it to input `JSON` string and use 
+ `from_json` to convert result output `JSON` string to `ARRAY`/`MAP`/`STRUCT` data.
+ - `COLLECTION ITEMS TERMINATED BY` and `MAP KEYS TERMINATED BY` are delimiters to split complex data such as
+  `ARRAY`/`MAP`/`STRUCT`, Spark use `to_json` and `from_json` to handle complex data type with `JSON` format, so 
+ `COLLECTION ITEMS TERMINATED BY` and `MAP KEYS TERMINATED BY` won't work in default row format.
+ - The standard output of the user script is treated as tab-separated `STRING` columns, any cell containing only `"\N"`
+ is re-interpreted as a `NULL` value, and then the resulting `STRING` column will be cast to the data type specified in `col_type`.
+ - If the actual number of output columns is less than the number of specified output columns,
+  insufficient output columns will be supplemented with `NULL`. 
+     ```
+     output tabs: `1, 2`
+     output columns: `A: INT, B INT, C: INT`
+     result: 
+       +---+---+------+
+       |  a|  b|     c|
+       +---+---+------+
+       |  1|  2|  NULL|
+       +---+---+------+
+     ```
+ - If the actual number of output columns is more than the number of specified output columns, 
+ the output columns only select the corresponding columns, and the remaining part will be discarded.
+ For example, output have three tabs and output columns only have two columns:
+     ```
+     output tabs: `1, 2, 3`
+     output columns: `A: INT, B INT`
+     result: 
+       +---+---+
+       |  a|  b|
+       +---+---+
+       |  1|  2|
+       +---+---+
+     ```
+ - If there is no `AS` clause after `USING my_script`, an output schema is `key: STRING, value: STRING`.
+ The `key` column contains all the characters before the first tab and the `value` column contains the remaining characters after the first tab.
+ If there are not enough tabs, Spark pads `NULL` value.
+      ```
+      output tabs: `1, 2, 3`
+      output columns: 
+      result: 
+        +-----+-------+
+        |  key|  value|
+        +-----+-------+
+        |    1|      2|
+        +-----+-------+
+   
+      output tabs: `1, 2`
+      output columns: 
+      result: 
+        +-----+-------+
+        |  key|  value|
+        +-----+-------+
+        |    1|   NULL|
+        +-----+-------+
+      ```
+
+### Hive SerDe behavior
+
+When enable Hive and use Hive Serde mode:
+ - Spark uses the Hive SerDe `org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe` by default, so columns are casted
+ to `STRING` and combined by tabs before feeding to the user script.
+ - All `NULL` values are converted to the literal string `"\N"` in order to differentiate `NULL` values from empty strings.
```

Comment:
Is that literally the 4-character string passed? with quotes? and not a new line but the two chars `\N`?

Reply:
Yea, you can see the screen shot

Reply:
![image](https://user-images.githubusercontent.com/46485123/126169722-a3a64a52-4213-4722-9fb7-0e7dea907c07.png)

Reply:
I'm asking whether it's correct - that seems surprising

Reply:
> I'm asking whether it's correct - that seems surprising

Below should be more clear?

All `NULL` values are converted to the literal string `'\N'` in order to differentiate `NULL` values from `'NULL'`.

Reply:
OK if it's definitely that 4 character string, that's fine.
BTW is it single or double quotes? your comment differs from the doc.

Reply:
> OK if it's definitely that 4 character string, that's fine.
> BTW is it single or double quotes? your comment differs from the doc.

I change it to single quotes since in SQL doc for string value we always use single quote?

Reply:
> OK if it's definitely that 4 character string, that's fine.
> BTW is it single or double quotes? your comment differs from the doc.

have not update the screan shot yet

Reply:
That's not my question. You are suggesting here that the value has four characters: a quote, a backlash, N, a quote. I'm not sure that's what you mean, as you are saying the quote is just a doc convention. If you are code-quoting the string, you are implying the whole thing is a literal.

Here's a simple question - how many characters does the actual value have?

Reply:
> That's not my question. You are suggesting here that the value has four characters: a quote, a backlash, N, a quote. I'm not sure that's what you mean, as you are saying the quote is just a doc convention. If you are code-quoting the string, you are implying the whole thing is a literal.
> 
> Here's a simple question - how many characters does the actual value have?

Got the point. like this 

 - All literal `NULL` values are converted to the literal string `\N` in order to differentiate literal `NULL` values from literal string `NULL`.

Reply:
Right, the literal should be `\N` then